### PR TITLE
Add dev container (IT-3844)

### PR DESCRIPTION
**Pull request by tschaffter**
## Description

See [IT-3844](https://sagebionetworks.jira.com/browse/IT-3844) for the goal and motivation.

## Changelogs

- Add the definition of a dev container that includes all the tools required by this project

### Build and open this project inside the dev container (VS Code)

1. Clone the repo
2. Open the repo folder in VS Code
3. VS Code should prompt you to open the project inside the dev container. If not, open the Command Palette and select "Dev Containers: Open Folder in Container..."
4. That's it! You are now inside the dev container and have access to all the development tools

### Tools

Inside the dev container:

```console
$ echo $SHELL
/bin/bash

$ lsb_release -a
No LSB modules are available.
Distributor ID: Ubuntu
Description:    Ubuntu 22.04.4 LTS
Release:        22.04
Codename:       jammy

$ node --version
v22.6.0

$ python --version
Python 3.12.0

$ aws --version
aws-cli/2.17.29 Python/3.11.9 Linux/5.15.133.1-microsoft-standard-WSL2 exe/x86_64.ubuntu.22

$ cdk --version
2.151.0 (build b8289e2)
```

### Synthetize the CDK app

```
vscode ➜ /workspaces/openchallenges-aws-cdk (add-dev-container) $ python3 -m venv .venv
vscode ➜ /workspaces/openchallenges-aws-cdk (add-dev-container) $ source .venv/bin/activate
(.venv) vscode ➜ /workspaces/openchallenges-aws-cdk (add-dev-container) $ pip install -r requirements.txt -r requirements-dev.txt
Collecting aws-cdk-lib==2.139.0 (from -r requirements.txt (line 1))
  Obtaining dependency information for aws-cdk-lib==2.139.0 from https://files.pythonhosted.org/packages/0b/b2/0683e1484ce9203f3dfbecdc2fe315cb34f7ff9790347f0f6b33fcb9cc3c/aws_cdk_lib-2.139.0-py3-none-any.whl.metadata
  Downloading aws_cdk_lib-2.139.0-py3-none-any.whl.metadata (59 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 59.3/59.3 kB 2.7 MB/s eta 0:00:00
Collecting constructs<11.0.0,>=10.0.0 (from -r requirements.txt (line 2))
  Obtaining dependency information for constructs<11.0.0,>=10.0.0 from https://files.pythonhosted.org/packages/8f/82/5b1407b9747a8c0133d56433dd9b67ec622558b4b47dad6b1751c9d5aeeb/constructs-10.3.0-py3-none-any.whl.metadata
  Downloading constructs-10.3.0-py3-none-any.whl.metadata (2.7 kB)
Collecting boto3>=1.34.1 (from -r requirements.txt (line 3))
  Obtaining dependency information for boto3>=1.34.1 from https://files.pythonhosted.org/packages/93/cb/963e249b98b03b5b078917faf34399b8a85aadcdd52d9f408de2e035b72c/boto3-1.34.160-py3-none-any.whl.metadata
  Downloading boto3-1.34.160-py3-none-any.whl.metadata (6.6 kB)
Collecting pytest==6.2.5 (from -r requirements-dev.txt (line 1))
  Obtaining dependency information for pytest==6.2.5 from https://files.pythonhosted.org/packages/40/76/86f886e750b81a4357b6ed606b2bcf0ce6d6c27ad3c09ebf63ed674fc86e/pytest-6.2.5-py3-none-any.whl.metadata
  Downloading pytest-6.2.5-py3-none-any.whl.metadata (7.5 kB)
Collecting aws-cdk.asset-awscli-v1<3.0.0,>=2.2.202 (from aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for aws-cdk.asset-awscli-v1<3.0.0,>=2.2.202 from https://files.pythonhosted.org/packages/72/67/3e15a7bd6aa67e6e136b09fec7fc3e21483c7d8ad6ce38649117ae25a4dd/aws_cdk.asset_awscli_v1-2.2.202-py3-none-any.whl.metadata
  Downloading aws_cdk.asset_awscli_v1-2.2.202-py3-none-any.whl.metadata (1.1 kB)
Collecting aws-cdk.asset-kubectl-v20<3.0.0,>=2.1.2 (from aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for aws-cdk.asset-kubectl-v20<3.0.0,>=2.1.2 from https://files.pythonhosted.org/packages/d9/97/b4eaa76c87a745ecf61b7ab49a712653e6f5c71f8bd476e1a0d5215b55d3/aws_cdk.asset_kubectl_v20-2.1.2-py3-none-any.whl.metadata
  Downloading aws_cdk.asset_kubectl_v20-2.1.2-py3-none-any.whl.metadata (1.2 kB)
Collecting aws-cdk.asset-node-proxy-agent-v6<3.0.0,>=2.0.3 (from aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for aws-cdk.asset-node-proxy-agent-v6<3.0.0,>=2.0.3 from https://files.pythonhosted.org/packages/52/12/9abbed9dcf7eb1d30563d0bd38d163b3e2386756d7fd0a9ef2575047b478/aws_cdk.asset_node_proxy_agent_v6-2.0.3-py3-none-any.whl.metadata
  Downloading aws_cdk.asset_node_proxy_agent_v6-2.0.3-py3-none-any.whl.metadata (1.1 kB)
Collecting jsii<2.0.0,>=1.97.0 (from aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for jsii<2.0.0,>=1.97.0 from https://files.pythonhosted.org/packages/f3/d1/c6d42a14049576dd4b0861ad4e76be95c0ce64523b114857f75c7d20ff0d/jsii-1.102.0-py3-none-any.whl.metadata
  Downloading jsii-1.102.0-py3-none-any.whl.metadata (79 kB)
     ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 79.7/79.7 kB 12.1 MB/s eta 0:00:00
Collecting publication>=0.0.3 (from aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for publication>=0.0.3 from https://files.pythonhosted.org/packages/f8/d3/6308debad7afcdb3ea5f50b4b3d852f41eb566a311fbcb4da23755a28155/publication-0.0.3-py2.py3-none-any.whl.metadata
  Downloading publication-0.0.3-py2.py3-none-any.whl.metadata (8.7 kB)
Collecting typeguard~=2.13.3 (from aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for typeguard~=2.13.3 from https://files.pythonhosted.org/packages/9a/bb/d43e5c75054e53efce310e79d63df0ac3f25e34c926be5dffb7d283fb2a8/typeguard-2.13.3-py3-none-any.whl.metadata
  Downloading typeguard-2.13.3-py3-none-any.whl.metadata (3.6 kB)
Collecting attrs>=19.2.0 (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for attrs>=19.2.0 from https://files.pythonhosted.org/packages/6a/21/5b6702a7f963e95456c0de2d495f67bf5fd62840ac655dc451586d23d39a/attrs-24.2.0-py3-none-any.whl.metadata
  Downloading attrs-24.2.0-py3-none-any.whl.metadata (11 kB)
Collecting iniconfig (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for iniconfig from https://files.pythonhosted.org/packages/ef/a6/62565a6e1cf69e10f5727360368e451d4b7f58beeac6173dc9db836a5b46/iniconfig-2.0.0-py3-none-any.whl.metadata
  Downloading iniconfig-2.0.0-py3-none-any.whl.metadata (2.6 kB)
Collecting packaging (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for packaging from https://files.pythonhosted.org/packages/08/aa/cc0199a5f0ad350994d660967a8efb233fe0416e4639146c089643407ce6/packaging-24.1-py3-none-any.whl.metadata
  Downloading packaging-24.1-py3-none-any.whl.metadata (3.2 kB)
Collecting pluggy<2.0,>=0.12 (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for pluggy<2.0,>=0.12 from https://files.pythonhosted.org/packages/88/5f/e351af9a41f866ac3f1fac4ca0613908d9a41741cfcf2228f4ad853b697d/pluggy-1.5.0-py3-none-any.whl.metadata
  Downloading pluggy-1.5.0-py3-none-any.whl.metadata (4.8 kB)
Collecting py>=1.8.2 (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for py>=1.8.2 from https://files.pythonhosted.org/packages/f6/f0/10642828a8dfb741e5f3fbaac830550a518a775c7fff6f04a007259b0548/py-1.11.0-py2.py3-none-any.whl.metadata
  Downloading py-1.11.0-py2.py3-none-any.whl.metadata (2.8 kB)
Collecting toml (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for toml from https://files.pythonhosted.org/packages/44/6f/7120676b6d73228c96e17f1f794d8ab046fc910d781c8d151120c3f1569e/toml-0.10.2-py2.py3-none-any.whl.metadata
  Downloading toml-0.10.2-py2.py3-none-any.whl.metadata (7.1 kB)
Collecting botocore<1.35.0,>=1.34.160 (from boto3>=1.34.1->-r requirements.txt (line 3))
  Obtaining dependency information for botocore<1.35.0,>=1.34.160 from https://files.pythonhosted.org/packages/d3/4c/3a5c98d0bf29156d923436d69a191454d11143005a73ffe8ad9ff43a6016/botocore-1.34.160-py3-none-any.whl.metadata
  Downloading botocore-1.34.160-py3-none-any.whl.metadata (5.7 kB)
Collecting jmespath<2.0.0,>=0.7.1 (from boto3>=1.34.1->-r requirements.txt (line 3))
  Obtaining dependency information for jmespath<2.0.0,>=0.7.1 from https://files.pythonhosted.org/packages/31/b4/b9b800c45527aadd64d5b442f9b932b00648617eb5d63d2c7a6587b7cafc/jmespath-1.0.1-py3-none-any.whl.metadata
  Downloading jmespath-1.0.1-py3-none-any.whl.metadata (7.6 kB)
Collecting s3transfer<0.11.0,>=0.10.0 (from boto3>=1.34.1->-r requirements.txt (line 3))
  Obtaining dependency information for s3transfer<0.11.0,>=0.10.0 from https://files.pythonhosted.org/packages/3c/4a/b221409913760d26cf4498b7b1741d510c82d3ad38381984a3ddc135ec66/s3transfer-0.10.2-py3-none-any.whl.metadata
  Downloading s3transfer-0.10.2-py3-none-any.whl.metadata (1.7 kB)
Collecting python-dateutil<3.0.0,>=2.1 (from botocore<1.35.0,>=1.34.160->boto3>=1.34.1->-r requirements.txt (line 3))
  Obtaining dependency information for python-dateutil<3.0.0,>=2.1 from https://files.pythonhosted.org/packages/ec/57/56b9bcc3c9c6a792fcbaf139543cee77261f3651ca9da0c93f5c1221264b/python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata
  Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl.metadata (8.4 kB)
Collecting urllib3!=2.2.0,<3,>=1.25.4 (from botocore<1.35.0,>=1.34.160->boto3>=1.34.1->-r requirements.txt (line 3))
  Obtaining dependency information for urllib3!=2.2.0,<3,>=1.25.4 from https://files.pythonhosted.org/packages/ca/1c/89ffc63a9605b583d5df2be791a27bc1a42b7c32bab68d3c8f2f73a98cd4/urllib3-2.2.2-py3-none-any.whl.metadata
  Downloading urllib3-2.2.2-py3-none-any.whl.metadata (6.4 kB)
Collecting attrs>=19.2.0 (from pytest==6.2.5->-r requirements-dev.txt (line 1))
  Obtaining dependency information for attrs>=19.2.0 from https://files.pythonhosted.org/packages/e0/44/827b2a91a5816512fcaf3cc4ebc465ccd5d598c45cefa6703fcf4a79018f/attrs-23.2.0-py3-none-any.whl.metadata
  Downloading attrs-23.2.0-py3-none-any.whl.metadata (9.5 kB)
Collecting cattrs<23.3,>=1.8 (from jsii<2.0.0,>=1.97.0->aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for cattrs<23.3,>=1.8 from https://files.pythonhosted.org/packages/b3/0d/cd4a4071c7f38385dc5ba91286723b4d1090b87815db48216212c6c6c30e/cattrs-23.2.3-py3-none-any.whl.metadata
  Downloading cattrs-23.2.3-py3-none-any.whl.metadata (10 kB)
Collecting importlib-resources>=5.2.0 (from jsii<2.0.0,>=1.97.0->aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for importlib-resources>=5.2.0 from https://files.pythonhosted.org/packages/75/06/4df55e1b7b112d183f65db9503bff189e97179b256e1ea450a3c365241e0/importlib_resources-6.4.0-py3-none-any.whl.metadata
  Downloading importlib_resources-6.4.0-py3-none-any.whl.metadata (3.9 kB)
Collecting typing-extensions<5.0,>=3.8 (from jsii<2.0.0,>=1.97.0->aws-cdk-lib==2.139.0->-r requirements.txt (line 1))
  Obtaining dependency information for typing-extensions<5.0,>=3.8 from https://files.pythonhosted.org/packages/26/9f/ad63fc0248c5379346306f8668cda6e2e2e9c95e01216d2b8ffd9ff037d0/typing_extensions-4.12.2-py3-none-any.whl.metadata
  Downloading typing_extensions-4.12.2-py3-none-any.whl.metadata (3.0 kB)
Collecting six>=1.5 (from python-dateutil<3.0.0,>=2.1->botocore<1.35.0,>=1.34.160->boto3>=1.34.1->-r requirements.txt (line 3))
  Obtaining dependency information for six>=1.5 from https://files.pythonhosted.org/packages/d9/5a/e7c31adbe875f2abbb91bd84cf2dc52d792b5a01506781dbcf25c91daf11/six-1.16.0-py2.py3-none-any.whl.metadata
  Downloading six-1.16.0-py2.py3-none-any.whl.metadata (1.8 kB)
Downloading aws_cdk_lib-2.139.0-py3-none-any.whl (35.1 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 35.1/35.1 MB 42.9 MB/s eta 0:00:00
Downloading pytest-6.2.5-py3-none-any.whl (280 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 280.7/280.7 kB 37.1 MB/s eta 0:00:00
Downloading constructs-10.3.0-py3-none-any.whl (58 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 58.2/58.2 kB 9.5 MB/s eta 0:00:00
Downloading boto3-1.34.160-py3-none-any.whl (139 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 139.2/139.2 kB 21.5 MB/s eta 0:00:00
Downloading aws_cdk.asset_awscli_v1-2.2.202-py3-none-any.whl (16.2 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 16.2/16.2 MB 48.6 MB/s eta 0:00:00
Downloading aws_cdk.asset_kubectl_v20-2.1.2-py3-none-any.whl (25.4 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 25.4/25.4 MB 31.0 MB/s eta 0:00:00
Downloading aws_cdk.asset_node_proxy_agent_v6-2.0.3-py3-none-any.whl (1.5 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 1.5/1.5 MB 40.2 MB/s eta 0:00:00
Downloading botocore-1.34.160-py3-none-any.whl (12.5 MB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 12.5/12.5 MB 41.4 MB/s eta 0:00:00
Downloading jmespath-1.0.1-py3-none-any.whl (20 kB)
Downloading jsii-1.102.0-py3-none-any.whl (554 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 554.7/554.7 kB 26.0 MB/s eta 0:00:00
Downloading attrs-23.2.0-py3-none-any.whl (60 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 60.8/60.8 kB 11.2 MB/s eta 0:00:00
Downloading pluggy-1.5.0-py3-none-any.whl (20 kB)
Downloading publication-0.0.3-py2.py3-none-any.whl (7.7 kB)
Downloading py-1.11.0-py2.py3-none-any.whl (98 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 98.7/98.7 kB 16.5 MB/s eta 0:00:00
Downloading s3transfer-0.10.2-py3-none-any.whl (82 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 82.7/82.7 kB 14.0 MB/s eta 0:00:00
Downloading typeguard-2.13.3-py3-none-any.whl (17 kB)
Downloading iniconfig-2.0.0-py3-none-any.whl (5.9 kB)
Downloading packaging-24.1-py3-none-any.whl (53 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 54.0/54.0 kB 9.4 MB/s eta 0:00:00
Downloading toml-0.10.2-py2.py3-none-any.whl (16 kB)
Downloading cattrs-23.2.3-py3-none-any.whl (57 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 57.5/57.5 kB 10.3 MB/s eta 0:00:00
Downloading importlib_resources-6.4.0-py3-none-any.whl (38 kB)
Downloading python_dateutil-2.9.0.post0-py2.py3-none-any.whl (229 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 229.9/229.9 kB 23.2 MB/s eta 0:00:00
Downloading typing_extensions-4.12.2-py3-none-any.whl (37 kB)
Downloading urllib3-2.2.2-py3-none-any.whl (121 kB)
   ━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━ 121.4/121.4 kB 20.8 MB/s eta 0:00:00
Downloading six-1.16.0-py2.py3-none-any.whl (11 kB)
Installing collected packages: publication, urllib3, typing-extensions, typeguard, toml, six, py, pluggy, packaging, jmespath, iniconfig, importlib-resources, attrs, python-dateutil, pytest, cattrs, jsii, botocore, s3transfer, constructs, aws-cdk.asset-node-proxy-agent-v6, aws-cdk.asset-kubectl-v20, aws-cdk.asset-awscli-v1, boto3, aws-cdk-lib
Successfully installed attrs-23.2.0 aws-cdk-lib-2.139.0 aws-cdk.asset-awscli-v1-2.2.202 aws-cdk.asset-kubectl-v20-2.1.2 aws-cdk.asset-node-proxy-agent-v6-2.0.3 boto3-1.34.160 botocore-1.34.160 cattrs-23.2.3 constructs-10.3.0 importlib-resources-6.4.0 iniconfig-2.0.0 jmespath-1.0.1 jsii-1.102.0 packaging-24.1 pluggy-1.5.0 publication-0.0.3 py-1.11.0 pytest-6.2.5 python-dateutil-2.9.0.post0 s3transfer-0.10.2 six-1.16.0 toml-0.10.2 typeguard-2.13.3 typing-extensions-4.12.2 urllib3-2.2.2

[notice] A new release of pip is available: 23.2.1 -> 24.2
[notice] To update, run: pip install --upgrade pip
(.venv) vscode ➜ /workspaces/openchallenges-aws-cdk (add-dev-container) $ cdk synth
Successfully synthesized to /workspaces/openchallenges-aws-cdk/cdk.out
Supply a stack id (openchallenges-buckets, openchallenges-network, openchallenges-ecs, openchallenges-mariadb, openchallenges-elasticsearch, openchallenges-thumbor, openchallenges-config-server, openchallenges-service-registry, openchallenges-zipkin, openchallenges-image-service, openchallenges-organization-service, openchallenges-api-gateway, openchallenges-app, openchallenges-load-balancer, openchallenges-api-docs, openchallenges-apex) to display its template.
```

[IT-3844]: https://sagebionetworks.jira.com/browse/IT-3844?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by zaro0508 — CHANGES_REQUESTED** [review]
could we add some docs to the README file for this?

**Comment by tschaffter** [comment]
@zaro0508 This is done in #31, which would be merged after this PR.

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "AWS CDK & Python Development Environment",
+  "image": "mcr.microsoft.com/devcontainers/base:ubuntu-22.04",
+  "features": {
```
Comment (zaro0508) [inline .devcontainer/devcontainer.json:4]: can we add setup for pre-commit?  something like this.. https://github.com/Opentrons/vscode-dev-container-template/blob/main/.devcontainer/devcontainer.json#L41

Comment (tschaffter) [reply .devcontainer/devcontainer.json:4]: It's up to you. Feel free to suggest an update (using GitHub suggestion tool) to the `postCreateCommand` command below and I'll approve it.

File: .devcontainer/devcontainer.json [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{
+  "name": "AWS CDK & Python Development Environment",
+  "image": "mcr.microsoft.com/devcontainers/base:ubuntu-22.04",
```
Comment (zaro0508) [inline .devcontainer/devcontainer.json:3]:
```suggestion
"postCreateCommand": "pip install --user -U pre-commit && pre-commit install-hooks",
```

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (tschaffter) [reply .devcontainer/devcontainer.json:4]: Is this the command you want to run automatically once the container has started?

```
pip install --user -U pre-commit && pre-commit install-hooks
```

The command completes successfully in the container (takes 1 min 15 s to complete)

File: .devcontainer/devcontainer.json (hunk 2 above)
Comment (tschaffter) [inline .devcontainer/devcontainer.json:3]: You are overwriting the existing `postCreateCommand` defined at the bottom of the file.

File: .devcontainer/devcontainer.json [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+  "features": {
+    "ghcr.io/devcontainers/features/node:1.5.0": {
+      "version": "22.6.0"
+    },
+    "ghcr.io/devcontainers/features/python:1.6.3": {
+      "version": "3.12.0"
+    },
+    "ghcr.io/devcontainers/features/aws-cli:1": {}
+  },
+  "postCreateCommand": "npm install -g aws-cdk@2.151.0",
```
Comment (zaro0508) [inline .devcontainer/devcontainer.json]:
```suggestion
  "postCreateCommand": ["pip install --user -U pre-commit && pre-commit install-hooks","npm install -g aws-cdk@2.151.0"],
```

File: .devcontainer/devcontainer.json (hunk 1 above)
Comment (zaro0508) [reply .devcontainer/devcontainer.json:4]: yes, it's for https://github.com/Sage-Bionetworks-IT/openchallenges/blob/dev/README.md#static-analysis

File: .devcontainer/devcontainer.json [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+  "features": {
+    "ghcr.io/devcontainers/features/node:1.5.0": {
+      "version": "22.6.0"
+    },
+    "ghcr.io/devcontainers/features/python:1.6.3": {
+      "version": "3.12.0"
+    },
+    "ghcr.io/devcontainers/features/aws-cli:1": {}
+  },
+  "postCreateCommand": "npm install -g aws-cdk@2.151.0 && pip install --user -U pre-commit && pre-commit install-hooks",
```
Comment (zaro0508) [inline .devcontainer/devcontainer.json]: why use `&&`?  doesn't postCreateCommand take a list of strings?  https://github.com/devcontainers/spec/blob/main/docs/specs/devcontainer-reference.md#merge-logic

File: .devcontainer/devcontainer.json (hunk 3 above)
Comment (tschaffter) [reply .devcontainer/devcontainer.json]: Done

**Review by zaro0508 — APPROVED** [review]

File: .devcontainer/devcontainer.json (hunk 4 above)
Comment (tschaffter) [reply .devcontainer/devcontainer.json]: It takes a string or an array of strings. The code is now using an array purely for esthetic reasons.

**Pull request merged**